platform_code/stm32cube/serial/micro_ros_transport.cpp

```diff
@@ -82,19 +82,19 @@ size_t platformio_transport_read(struct uxrCustomTransport *transport,
   DMAStream *stream = (DMAStream *)transport->args;
 
   int ms_used = 0;
-  do {
+  while (true) {
     __disable_irq();
     rtail = stream->rbuffer_size - __HAL_DMA_GET_COUNTER(stream->uart->hdmarx);
     __enable_irq();
 
     uint16_t data_available =
         rhead <= rtail ? rtail - rhead : stream->rbuffer_size - rhead + rtail;
 
-    if (data_available >= len) break;
+    if (data_available >= len || ms_used >= timeout) break;
```

Comment:
This should not wait until read data is >= len. This should wait until there is available 1 o more Bytes.

Reply:
fixed it

Reply:
Cool, thanks